scripts/scripts.js

```diff
@@ -531,21 +565,64 @@ async function decoratePage(win = window) {
   const $main = doc.querySelector('main');
   if ($main) {
     decorateMain($main);
-    doc.querySelector('body').classList.add('appear');
-    setLCPTrigger(doc, async () => {
-      // post LCP actions go here
-
-      /* load gnav */
-      const header = document.querySelector('header');
-      header.setAttribute('data-block-name', 'gnav');
-      header.setAttribute('data-gnav-source', '/blog/gnav');
-      loadBlock(header);
-
-      await loadBlocks($main);
-      loadCSS('/styles/lazy-styles.css');
-      addFavIcon('/styles/favicon.svg');
+    getLCPCandidate((lcpCandidateEl) => {
+      setLCPTrigger(lcpCandidateEl, async () => {
+        // post LCP actions go here
+
+        /* load gnav */
+        const header = document.querySelector('header');
+        header.setAttribute('data-block-name', 'gnav');
+        header.setAttribute('data-gnav-source', '/blog/gnav');
+        loadBlock(header);
+
+        await loadBlocks($main);
+        loadCSS('/styles/lazy-styles.css');
+        addFavIcon('/styles/favicon.svg');
+      });
     });
+    document.querySelector('body').classList.add('appear');
   }
 }
 
 decoratePage(window);
+
+function stamp(message) {
+  if (window.name.includes('performance')) {
+    // eslint-disable-next-line no-console
+    console.log(`${new Date() - performance.timing.navigationStart}:${message}`);
+  }
+}
+
+stamp('start');
+
+function registerPerformanceLogger() {
+  try {
+    const polcp = new PerformanceObserver((entryList) => {
+      const entries = entryList.getEntries();
+      stamp(JSON.stringify(entries));
+      // eslint-disable-next-line no-console
+      console.log(entries[0].element);
+    });
+    polcp.observe({ type: 'largest-contentful-paint', buffered: true });
+
+    const pols = new PerformanceObserver((entryList) => {
+      const entries = entryList.getEntries();
+      stamp(JSON.stringify(entries));
+      console.log(entries[0].sources[0].node);
```

Comment:
do we need an eslint disable statement here?

Reply:
i agree that this whole performance instrumentation feels odd, but honestly i found it the most useful way to understand what's going on. everything else was either too slow, too detailed or not detailed enough. i think we should refactor this and put it somewhere in the project setup, the problem that i see is as this needs to be registered very early on i don't think it can be loaded externally, but would be worth a shot.